vignette addition and dependency help update

diegobilski opened the pull request:
Adds vignette on updating scripts to rsyncrosim v2. 
Changes ` rsyncrosim` to rsyncrosim on vignette titles. 
Change on dependency function help.

diegobilski committed:
vignette addition and dependency help update

Adds vignette on updating scripts to rsyncrosim v2.
Change on dependency function help.

katieb1 commented on R/dependency.R:
```diff
@@ -17,19 +17,13 @@ NULL
 #' Otherwise, returns list of saved or error messages for each dependency of each 
 #' scenario.
 #'
-#' Note that the order of dependencies can be important - dependencies added most 
```
I think you might have overwritten the updated docs for the dependencies. I would revert your changes to this script and then just make the one tweak to the priority wording and re-commit.

diegobilski committed:
a07 vignette update

Added core_ datasheets name change

katieb1 committed:
Update a07_updating_rsyncrosim_scripts_to_v2.Rmd

Add some headers and some edits for clarity

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -0,0 +1,117 @@
+
+## Library
+
+`info(ssimLibrary)` no longer shows the *Package name*, *Current package version* and *Minimum package version*, and the Input and Output folder paths have been combined into the *Data folder*.
```
Add code chunk with output to show update to `info()` function. Output could be added in a "hacky" way for clarity.

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
````diff
@@ -0,0 +1,117 @@
+```
+
+## Library
+
````
Add overview sentence about changes to library functions

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
````diff
@@ -0,0 +1,117 @@
+myLibrary <- ssimLibrary(packages = c())
+```
+
+The `addon` and `template` arguments have also been removed.
````
I would add this to above because it is unclear it relates to the `ssimLibrary()` function

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -0,0 +1,117 @@
+
+`info(ssimLibrary)` no longer shows the *Package name*, *Current package version* and *Minimum package version*, and the Input and Output folder paths have been combined into the *Data folder*.
+
+The `package` argument in the `ssimLibrary(package)` function has been renamed to `packages`
```
consistency in punctuation

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
````diff
@@ -0,0 +1,117 @@
+The `package` argument in the `ssimLibrary(package)` function has been renamed to `packages`
+
+```{r library, eval=FALSE}
+myLibrary <- ssimLibrary(packages = c())
````
I would show the old way of doing things versus the new way for comparison

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -0,0 +1,117 @@
+
+## Dependencies
+
+To view the existing dependencies for a scenario, use:
```
Add overview sentence on changes to dependencies. Add old way of doing things versus new way to highlight differences.

katieb1 replied:
Also I would generate some output to show what happens when you add dependencies

katieb1 replied:
Maybe something like:

> Scenario dependencies are now set using the `<-` operator in rsyncrosim. See below for examples on how the usage of the `dependency()` function has changed.

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -0,0 +1,117 @@
+
+## Multiprocessing
+
+The `run()` function no longer has the `jobs` argument. Use the `core_Multiprocessing` datasheet to set the number of cores to use. Note that this will affect every scenario run.
```
I think this part needs more detail. For instance, it's unclear that the `core_Multiprocessing` datasheet is set at the library scope. Also include an example of how to set the `core_Multiprocessing` datasheet.

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -0,0 +1,117 @@
+
+## Variable naming
+
+The **ID** in variable names have been changed to **Id**. It is generally safe to substitute `ID` for `Id` throughout your script, unless that affects any variables not related to SyncroSim.
```
I think this also needs more detail. It's specifically the "ID" in the primary key that has been updated. Some datasheets might have additional columns that have "ID" in them that the package developer has set. Give example

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -0,0 +1,117 @@
+
+## Core datasheets
+
+The `corestime_` datasheets are now simply called `core_`.
```
Add link to "System Datasheets" reference guide on syncrosim docs. More details on what the `core_` means (give a brief summary based on content in the System Datasheets reference guide overview).

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -0,0 +1,117 @@
+
+The `corestime_` datasheets are now simply called `core_`.
+
+## stsimsf
```
Remove section on stsimsf - this should go in the ST-Sim docs instead

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -0,0 +1,117 @@
+The ST-Sim Stock & Flow package, `stsimsf` have been incorporated into ST-Sim. Substitute all `stsimsf_` datasheets throghout your script for `stsim_`.
+The only exception is the `stsimsf_OutputOptions` that is now called `stsim_OutputOptionsStockFlow`.
+
+## Other removed functions
```
Move up above with other functions that have been removed (I created a section called "Deprecated functions").

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
````diff
@@ -0,0 +1,117 @@
+updatePackage()
+```
+
+## Other NEW functions
````
Move up above as well and give each new function it's own subsection. Write a brief description about each of these new functions.

katieb1 reviewed (changes requested):
Thanks, Diego! Looks like all the content is there, I think we just need to expand a bit to add more details on what is being updated and add more examples and comparisons with the old code. Could be nice to have an example library that you run through the vignette to show the output of the updated/new functions. 

I think also the dependencies docstrings need to be fixed! See my comment

diegobilski committed:
changes to vignette and dependency

additions to the Updating rsyncrosim scripts vignette.
changes to dependency function desctription

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
 disableAddon()
 enableAddon()
+
+datasheetRaster() # previously deprecated function. Use datasheetSpatRaster() instead.
```
This seems a bit unrelated to the above text to include these other functions in the same code chunk. I think each group of deprecated functions deserves a little bit of text explaining why they are no longer needed.

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
+library(rsyncrosim)
+mySession <- session()
+myLibrary <- ssimLibrary("Changes to rsyncrosim v2.ssim", packages=c("stsim", "helloworldSpatial"))
+scenario1 <- scenario(myLibrary, scenario = "Scenario 1")
```
I'm not sure why this other code is in here that doesn't have to do with the `ssimLibrary()` changes?

diegobilski replied:
this is just to show the addition of two packages, plus creating a couple of scenarios to be used later

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
-`info(ssimLibrary)` no longer shows the *Package name*, *Current package version* and *Minimum package version*, and the Input and Output folder paths have been combined into the *Data folder*.
-
-The `package` argument in the `ssimLibrary(package)` function has been renamed to `packages`
+Given this added flexibility, the `addon` argument has been removed from the `ssimLibrary()` function, and the `package` argument has been renamed to `packages`.
```
Maybe something more like:

> Now that addon packages no longer exist and SyncroSim libraries can support multiple packages, the following changes have been made to the `ssimLibrary()` function:
> * The `addon` argument has been removed.
> * The `template` argument has been removed. This is a temporary change, but in the meantime you can download template libraries from<a href="https://cloud.syncrosim.com/" target="_blank">SyncroSim Cloud</a>.
> * The `package` argument has been renamed to `packages` (plural).

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
-The `addon` and `template` arguments have also been removed.
+`info(ssimLibrary)` no longer shows the *Package name*, *Current package version* and *Minimum package version*, and the Input and Output folder paths have been combined into the *Data folder*. See the <a href="https://syncrosim.github.io/rsyncrosim/articles/a01_rsyncrosim_vignette_basic.html#access-model-metadata" taret="_blank">Introduction to rsyncrosim</a> for details.
+
+The `template` and `addon` arguments have also been removed. You can download template libraries from <a href="https://cloud.syncrosim.com/" target="_blank">SyncroSim Cloud</a>.
```
I would remove this line - it is referring to changes to `ssimLibrary()` but you bring up changes to `info(ssimLibrary)` in between so it is confusing

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
 
 ## Dependencies
 
 To view the existing dependencies for a scenario, use:
```
Make structure of this sentence consistent with below

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
````diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
 ```{r set dependencies, eval=FALSE}
-dependency(myScenario) <- c("Scenario 1", "Scenario 2")
-```
+# Change from:
````
I would make the comments consistent across sections when comparing old code to new code. So, when demonstrating old code, could be:

> `# rsyncrosim version 1:`

and new code could be:

> `# rsyncrosim version 2:`

katieb1 commented on vignettes/a07_updating_rsyncrosim_scripts_to_v2.Rmd:
```diff
@@ -37,78 +37,107 @@ Note that when you remove a package from your library any associated datasheets
+multiprocessing <-
+    data.frame(EnableMultiprocessing = TRUE,
+               MaximumJobs = 6,
+               EnableMultiScenario = TRUE) # optional, to run multiple scenarios in parallel
```
I would remove this argument - users couldn't set this from the `run()` function before anyway

diegobilski committed:
a07 vignette changes

incorporating suggested changes

katieb1 committed:
Clean up / reorganization of update vignette
